src/argus_htmx/destinations/forms.py

```diff
@@ -0,0 +1,79 @@
+from django import forms
+from django.forms import ModelForm
+
+from argus.notificationprofile.models import DestinationConfig, Media
+from argus.notificationprofile.serializers import RequestDestinationConfigSerializer
+from argus.notificationprofile.media import api_safely_get_medium_object
+
+
+class DestinationFormCreate(ModelForm):
+    settings = forms.CharField(required=True)
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request", None)
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = DestinationConfig
+        fields = ["label", "media", "settings"]
+        labels = {
+            "label": "Name",
+        }
+
+    def clean(self):
+        super().clean()
+        settings_key = _get_settings_key_for_media(self.cleaned_data["media"])
+        self.cleaned_data["settings"] = {settings_key: self.cleaned_data["settings"]}
+        self._init_serializer()
+        return self._validate_serializer()
+
+    def save(self):
+        # self.serializer should be initiated and validated in clean() before save() is called
+        self.serializer.save(user=self.request.user)
+
+    def _init_serializer(self):
+        serializer = RequestDestinationConfigSerializer(
+            data={
+                "media": self.cleaned_data["media"],
+                "label": self.cleaned_data.get("label", ""),
+                "settings": self.cleaned_data["settings"],
+            },
+            context={"request": self.request},
+        )
+        self.serializer = serializer
+
+    def _validate_serializer(self):
+        media = self.cleaned_data["media"]
+        settings_key = _get_settings_key_for_media(media)
+
+        # Add error messages from serializer to form
+        if not self.serializer.is_valid():
+            for error_name, error_detail in self.serializer.errors.items():
+                if error_name in ["media", "label", settings_key]:
+                    if error_name == settings_key:
+                        error_name = "settings"
+                    self.add_error(error_name, error_detail)
+                    # Serializer might add more data to the JSON dict
+                    if settings := self.serializer.data.get("settings"):
+                        self.cleaned_data["settings"] = settings
+        else:
+            # Serializer might add more data to the JSON dict
+            if settings := self.serializer.validated_data.get("settings"):
+                self.cleaned_data["settings"] = settings
+
+        if label := self.cleaned_data["label"]:
+            filter = DestinationConfig.objects.filter(label=label)
+            if self.instance:
+                filter = filter.exclude(pk=self.instance.pk)
+            if filter.exists():
+                self.add_error("label", "Name must be unique per media")
```

Comment:
If label is `""` then this wont work

Reply:
But it might be fine? `""` seems to get translated to None when the object is saved to the database, and the constraints seem to allow multiple destinations without a label.

But the thing that should be changes is `self.cleaned_data["label"] `should be `self.cleaned_data.get("label")`. This doesnt cause any issues when testing, because the frontend sends an empty string for label if the input box is empty, but since this form doesnt actually require label to be set, we should assume that there are situations where it matters.
```suggestion
        if label := self.cleaned_data.get("label"):
            filter = DestinationConfig.objects.filter(label=label)
            if self.instance:
                filter = filter.exclude(pk=self.instance.pk)
            if filter.exists():
                self.add_error("label", "Name must be unique per media")
```

Reply:
Additionally, it might be wrong to check the uniqueness of the name here in the form validator. There is already a constraint in the DestinationConfig model for exactly this. This could then be handled by trying to save and see if the constraint throws an exception and take it from there. This is more or less a repetition of that constraint. Buut you would need some code anyway, and it might be easier to just let the form handle everything, instead of having to catch the constraint exception and cram errors into the form from the outside